### PR TITLE
Security patches for libtiff 4.1.0

### PR DESCRIPTION
**Pull request by AndrewVallette**
## Changes
- Added 13 security patches from Debian
- Backported fix for CVE-2022-2867/CVE-2022-2868/CVE-2022-2869 from 4.4.0-rc1 
- Removed one patch which was no longer needed
- Updated URLs in about section

## Notes
Over a dozen security patches were incorporated from Debian: https://sources.debian.org/patches/tiff/4.1.0+git191117-2~deb10u4/

Additionally, critical CVEs CVE-2022-2867, CVE-2022-2868, and CVE-2022-2869 were addressed by backporting a fix from 4.4.0-rc1. This is found in patch file `0015_CVE-2022-2869.patch`.

The backport was created from the commit located here: https://gitlab.com/libtiff/libtiff/-/commit/07d79fcac2ead271b60e32aeb80f7b4f3be9ac8c

The procedure to recreate the buffer overflow from the CVE is located here: https://gitlab.com/libtiff/libtiff/-/issues/352

After building for `linux-64`, the above instructions were performed with the attached `poc.zip` file. No segmentation fault was observed, indicating that CVEs 2022-2867/8/9 were successfully patched.

### CODE REVIEW

**Review by varlackc — COMMENTED** [review]
We need to add the `License_family` to the recipe
https://github.com/AnacondaRecipes/libtiff-feedstock/blob/e94aff3087c9f20125a5e35410dfb4464865a39c/recipe/meta.yaml#L74&L75

File: recipe/meta.yaml [hunk 1]
```diff
@@ -56,16 +69,18 @@ test:
...
-  license_family: OTHER
-  summary: 'Support for the Tag Image File Format (TIFF).'
+  summary: Support for the Tag Image File Format (TIFF).
   description: |
     This software provides support for the Tag Image File Format (TIFF), a
     widely used format for storing image data.
-  dev_url: https://gitlab.com/libtiff/libtiff
+  # HTTPS does not work.
   doc_url: http://www.simplesystems.org/libtiff/
+  doc_source_url: https://gitlab.com/libtiff/libtiff/-/blob/master/README.md
```
Comment (marcoesters) [inline recipe/meta.yaml]: `https://gitlab.com/libtiff/libtiff/blob/master/README.md` goes to the same page and looks cleaner.

**Comment by AndrewVallette** [comment]
Thanks guys. I also noticed that there is an SPDX license type specifically for `libtiff`, which is called.. `libtiff`. So I have also updated that.

**Review by katietz — APPROVED** [review]
lgtm, thanks

**Pull request merged**